velox/adapters/s3fs/S3FileSystem.cpp

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/adapters/s3fs/S3FileSystem.h"
+#include "velox/common/file/File.h"
+#include "velox/core/Context.h"
+
+#include <fmt/format.h>
+#include <glog/logging.h>
+#include <memory>
+#include <mutex>
+#include <stdexcept>
+
+#include <aws/core/Aws.h>
+#include <aws/core/Region.h>
+#include <aws/core/auth/AWSCredentialsProviderChain.h>
+#include <aws/core/http/HttpResponse.h>
+#include <aws/core/utils/logging/ConsoleLogSystem.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/GetObjectRequest.h>
+
+namespace facebook::velox {
+
+constexpr std::string_view kSep{"/"};
+constexpr std::string_view s3Scheme{"s3:"};
+
+bool inline isS3File(const std::string_view filename) {
+  return (filename.substr(0, s3Scheme.size()) == s3Scheme);
+}
+
+inline Aws::String getAwsString(const std::string& s) {
```

Comment:
why not taking a string view here as well since you don't need a full blown std::string?

Reply:
I am passing the string by reference anyway. So I felt it wouldn't matter much.